src/pages/dar_application/collaborator/CollaboratorErrors.js

```diff
@@ -0,0 +1,57 @@
+import { div, h } from 'react-hyperscript-helpers';
+import { isEmpty, isNil } from 'lodash/fp';
+import { Notification } from '../../../components/Notification';
+
+export const computeCollaboratorErrors = (collaborator, showApproval) => {
+  const errors = [];
+
+  if (isNil(collaborator.name) || collaborator.name === '') {
+    errors.push('Must specify the name of the collaborator');
+  }
+
+  if (isNil(collaborator.eraCommonsId) || collaborator.eraCommonsId === '') {
+    errors.push('Must specify the eRA Commons ID of the collaborator');
+  }
+
+  if (isNil(collaborator.title) || collaborator.title === '') {
+    errors.push('Must specify the title of the collaborator');
+  }
+
+  if (isNil(collaborator.email) || collaborator.email === '') {
+    errors.push('Must specify the email of the collaborator.');
+  } else {
+    var testEmail = /\S+@\S+\.\S+/.test(collaborator.email);
+    if(testEmail === false) errors.push('The email must be a valid email.');
```

Comment:
Look at `utils.isEmailAddress` for a standard way of testing the validity of an email address. We should be using this single utility call so we aren't doing the same thing in different ways.

Reply:
See also `formValidators` for general ways to validate user input in a form; I've been using those functions in other places, might be a good central place for stuff like this. There is already an email validator, too.